astropy/units/format/utils.py

```diff
@@ -245,3 +253,12 @@ def unit_deprecation_warning(
     if decomposed is not None:
         message += f" Suggested: {decomposed}."
     warnings.warn(message, UnitsWarning)
+
+
+def _get_non_keyword_units(
+    bases: Iterable[str], prefixes: Sequence[str]
+) -> Generator[tuple[str, str], None, None]:
+    for base in bases:
+        for prefix in prefixes:
+            if not iskeyword(unit := prefix + base):
+                yield unit, base
```

Comment:
Just a suggestion to collapse the nested loop using the Cartesian product.

```suggestion
    for base, prefix in itertools.product(bases, prefixes):
        if not iskeyword(unit := prefix + base):
            yield unit, base
```

Reply:
We could save one line here, but we'd need an extra line to import `itertools`, so it's not clear to me that would be a good tradeoff.

Reply:
I slightly prefer the explicitness of what you have.